src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -547,7 +547,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			// for keyboard events that resolve to commands described
 			// within commandsToSkipShell, either alert or skip processing by xterm.js
 			if (resolveResult && this._skipTerminalCommands.some(k => k === resolveResult.commandId) && !this._configHelper.config.sendKeybindingsToShell) {
-				if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true)) {
+				if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true) &&
+					resolveResult.commandId !== 'workbench.action.terminal.toggleTerminal' &&
```

Comment:
A possible issue with this. If I have focus in the terminal, and I press a keybinding bound to one of these, and I expect it to go to the terminal but it triggers the command. That's actually the exact scenario we are trying to capture.